### PR TITLE
First question

### PR DESCRIPTION
**Pull request by NatalijaGucevska**

### CODE REVIEW

File: AssignmentOne.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+public class AssignmentOne {
+
+	public static void main(String[] args) {
+		Q1 q1 = new Q1();
+		
+		System.out.println("Expected false, returned " + q1.isPermutation(null, null));
+		System.out.println("Expected false, returned " + q1.isPermutation("", null));
+		System.out.println("Expected false, returned " + q1.isPermutation(null, "HelLo"));
+		System.out.println("Expected false, returned " + q1.isPermutation("Hello", "Helo"));
+		System.out.println("Expected true, returned " + q1.isPermutation("Hello", "HelLo"));
```
Comment (Thomas-de-Rivaz) [inline AssignmentOne.java:11]: Also a test case where the letters are shuffled would be good, "Hello" vs "elloh"

File: Q1.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+				return false;
+			} else {
+				if(val == 1) {
+					usedLetters++;
+				}
+				occurrences.put(c, val - 1);	
+			}
+		}
+		
+		return usedLetters == occurrences.size();
```
Comment (Thomas-de-Rivaz) [inline Q1.java]: Do you need to count usedLetters? Or can you just return true at this point.

Given that your strings have already been checked for the same length, and any missing characters should already have caused line 31 to exit.

**Review by Thomas-de-Rivaz — APPROVED** [review]
Good test cases.

File: Q1.java (hunk 2 above)
Comment (NatalijaGucevska) [reply Q1.java]: Indeed, it is unnecessary verification. Thank you.

**Pull request merged**